SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamPoller.java

```diff
@@ -0,0 +1,37 @@
+package com.hubspot.singularity.scheduler;
+
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hubspot.singularity.SingularityAction;
+import com.hubspot.singularity.config.SingularityConfiguration;
+import com.hubspot.singularity.data.DisasterManager;
+
+@Singleton
+public class SingularityUpstreamPoller extends SingularityLeaderOnlyPoller {
```

Comment:
Last thing for this class, there is an `abortsOnError` method in `SingularityLeaderOnlyPoller`. We should override it to return false for this. For some of the leader only pollers that have to do with task state, singularity aborts + restarts to preserve state integrity. That is not necessary for this poller so we can add:
```java
  @Override
  protected boolean abortsOnError() {
    return false;
  }
```